### ISSUE TITLE
[Druide] - Cat Form! (BB #456)

### ISSUE THREAD

**Issue by OregonAsari** [OregonCore/OregonCore#456]
This issue was migrated from bitbucket.
**Original Reporter:** ScullyNY
**Original Date:** 29.04.2013 14:06:36 GMT+0000
**Original Priority:** critical
**Original Type:** bug
**Original State:** resolved
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/456

<hr>

You can't use CatForm from Druid in Instances!
When you cast CatForm outsite from the instance and join, you're too in CatForm.

When u use it inside from the instance - don't change your form. (cast  and no change!)
You cant make dmg at Cat.. Druid DD can't play! Hard Bug!

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** ScullyNY
**Original Date:** 29.04.2013 14:07:38 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/456/#comment-3907715

<hr>

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** ScullyNY
**Original Date:** 29.04.2013 14:08:06 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/456/#comment-3907726

<hr>

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zaffy
**Original Date:** 29.04.2013 17:03:05 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/456/#comment-3910061

<hr>

(sorry this wasnt my comment :) someone played with my computer)

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** bieleib92
**Original Date:** 01.05.2013 14:08:54 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/456/#comment-3930358

<hr>

Test last rev Trinity Core 2.4.3 and we'll know when this issue appeared (if it's a TC or OC problem).

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** cca
**Original Date:** 05.05.2013 19:51:49 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/456/#comment-3968499

<hr>

Confirmed.
Old MaNGOS bug...

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** release_ok
**Original Date:** 06.05.2013 06:37:05 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/456/#comment-3971487

<hr>

vmap.enableindoorcheck = 0 - and your issue will be fixed: hacky but working.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** ScullyNY
**Original Date:** 06.05.2013 10:52:25 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/456/#comment-3973982

<hr>

this is not a final solution of the problem.. its not bug fix ;)

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** release_ok
**Original Date:** 06.05.2013 12:03:55 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/456/#comment-3974688

<hr>

Ye, thats why i haven't closed issue.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** bieleib92
**Original Date:** 07.05.2013 00:34:54 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/456/#comment-3981772

<hr>

cca: is this fixed in BC now? i've seen that in the forum, but i don't see any fix in commits. 

Thanks ;)

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** cca
**Original Date:** 07.05.2013 18:55:20 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/456/#comment-3992096

<hr>

In fact, I believe that this bug was a result of another error (communication with the cliente).
Should be working in BC, but I need to confirm with tests.
Do not use the commits as a reference. I do not separate the fixes.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** bieleib92
**Original Date:** 08.05.2013 00:33:29 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/456/#comment-3996522

<hr>

Thanks for the info! I'll test it

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 24.06.2013 02:41:20 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/456/#comment-4944480

<hr>

Below is a working fix, albeit a hacky one. This'll enable druid cat form indoors, you can change the config back to 1.
**_Changes made to Spell.cpp**_

```
#!diff

         VMAP::VMapFactory::createOrGetVMapManager()->isLineOfSightCalcEnabled())
     {
         if (m_spellInfo->Attributes & SPELL_ATTR_OUTDOORS_ONLY &&
-                !m_caster->GetMap()->IsOutdoors(m_caster->GetPositionX(), m_caster->GetPositionY(), m_caster->GetPositionZ()))
+                !m_caster->GetMap()->IsOutdoors(m_caster->GetPositionX(), m_caster->GetPositionY(), m_caster->GetPositionZ())
+               && m_spellInfo->Id != 768 && m_spellInfo->Id != 24866)
             return SPELL_FAILED_ONLY_OUTDOORS;

         if (m_spellInfo->Attributes & SPELL_ATTR_INDOORS_ONLY &&
```

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** release_ok
**Original Date:** 25.01.2014 10:45:34 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/456/#comment-7917760

<hr>

Resolved issue #456 - Cat Form

Some spells like Cat Form didn't work correctly if vmap outdoor checks were enabled.
Cat form was unable to even use in indoors because its auras were cancelling it.
The aura was Feral Swiftness. Even though it could be hackfixed, it still wouldn't work as expected.
With hackfix it works like this:
    - ferel swiftness works also in indoors (if casted indoors)
    - feral swiftness will be cancelled when one comes from outdoors to indoors
The feral swiftness is required to work only outside.

With this fix it works like it should have:
    - if cat form is casted indoors, feral swiftness is not applied
    - if you casted cat form indoors and come outdoors, FS will be applied automatically
    - if you casted cat form outdoors the FS buff is applied automatically,
      then if you enter indoors, it will be cancelled, and if you return outdoors,
      it will be applied again.
This fix is not a hackfix but (I think) a good way to do this also for other spells but I can't recall any of them.

→ <<cset aada2485c305>>
